src/core/annotation.js

```diff
@@ -945,6 +974,10 @@ class PopupAnnotation extends Annotation {
     this.data.parentType = isName(parentSubtype) ? parentSubtype.name : null;
     this.data.parentId = dict.getRaw('Parent').toString();
     this.data.title = stringToPDFString(parentItem.get('T') || '');
+    this.data.creationDate = isString(parentItem.get('CreationDate')) ?
+                             parentItem.get('CreationDate') : null;
+    this.data.modificationDate = isString(parentItem.get('M')) ?
+                                 parentItem.get('M') : null;
```

Comment:
Wouldn't it be much better to handle these properties in the same way as the `C` one below instead? (That would also get rid of the double `Dict` lookups here.)

Reply:
Fixed in the new commit. I also added a unit test for the `C` one below since that was missing.